helm/kanister-operator/values.yaml

```diff
@@ -13,6 +13,10 @@ serviceAccount:
 controller:
   service:
     port: 443
+  # createCRDs specifies if kanister controller should create the CRDs
+  # false : CRDs would be created by helm
+  # true : CRDs would be created by kanister controller
+  createCRDs: false
```

Comment:
So, even if we set the `createCRDs` field in the `values.yaml` to be true, our controller is going to update the CRDs which should be ok I think.

Reply:
> That's assuming the controller has the required RBAC permissions to manage CRDs, right? Looking back at the slack [convo](https://kanisterio.slack.com/archives/C85C8V22J/p1639747585010400), I thought the issue started because users changed the controller's RBAC to not be able to manage CRDs.

Controller not having authz to manage the CRDs is the case where people are going to use `createCRDs : false`. And in that case helm will be responsible to create the CRDs.

> Also, I am more interested in how the default of createCRDs: false will affect existing users where their existing CRDs aren't managed by Helm. On their next helm upgrade, Helm might fail due to conflicting existing resources.

To test this I  installed released (latest) version of kanister and then upgraded that to locally chnaged chart with changes  `createCRDs :  false` and it was upgraded successfully without failure.

Reply:
Thanks for confirming the behaviour of `createCRDs : false`. The inclusion of the CRD rules in `helm/kanister-operator/templates/rbac.yaml` confused me earlier.

Also, thanks for testing the upgrade using the latest version. Just out of curiosity, does that mean we can remove the controller-generated CRDs completely? This PR is going to default to using the Helm-managed CRDs anyway, and upgrade seems to work.

Reply:
Hi @ihcsim,
Like we discussed in this thread https://kanisterio.slack.com/archives/C85C8V22J/p1644962226053899 we need to have `crds` in helm template as well controller independently should also be able to create CRDs for the cases where kanister is not being consumed as sub chart.

Reply:
One additional consideration is that I don't believe Helm will update the CRDs when they change. However, the controller's code will create **or update** the CRDs.

With this in mind, perhaps calling the option `updateCRDs` instead might make more sense.

It seems like this would make it more obvious how a consumer could adopt updates to the CRDs versus manually installing the updates (or knowing that they need to set `createCRDs=true` even though the CRDs are already installed).

Reply:
thanks Mark.